extension/observer/hostobserver/extension.go

```diff
@@ -29,7 +29,7 @@ import (
 )
 
 type hostObserver struct {
-	observer.EndpointsWatcher
+	*observer.EndpointsWatcher
```

Comment:
What was the reason for making this a pointer?

Reply:
There's a fair amount of state in `EndpointsWatcher` and its zero value didn't seem that relevant to me.

Reply:
I guess the parent observers are passed and treated by reference so there's less concern of copying the EndpointsWatcher by embedding the struct. Is there a reason to do so?

Reply:
There isn't anything that I would call out to embed the struct directly, the only reason I would personally choose a non pointer over a pointer was to ensure that the zero values are set and if by chance I did forget to set something, it should error instead of panic.

Reply:
I think if adopters embedded the struct then to better prevent state copying `EndpointsWatcher`'s fields should become pointers, which increases the likelihood of nil pointer dereferencing. By providing a `NewEndpointsWatcher` ~constructor in these changes, my aim was to make adoption a bit easier considering there are no other requirements to implement an `Observable`.

Reply:
Makes sense to me :D